src/components/Map.js

```diff
@@ -29,9 +35,9 @@ class Map extends Component {
     });
   }
 
-  shouldComponentUpdate(nextProps) {
+  componentDidUpdate(nextProps) {
+    let map = this.state.map;
```

Comment:
Surprised the linter didn't suggest const for this because map is not reassigned.